src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -453,6 +462,50 @@ class _BuildPexRequirementsSetup:
     concurrency_available: int
 
 
+@dataclass(frozen=True)
+class ReqStrings:
+    req_strings: tuple[str, ...]
+
+
+@rule
+async def get_req_strings(pex_reqs: PexRequirements) -> ReqStrings:
+    addrs: list[Address] = []
+    specs: list[str] = []
+    req_strings: list[str] = []
+    for req_str_or_addr in pex_reqs.req_strings_or_addrs:
+        if isinstance(req_str_or_addr, Address):
+            addrs.append(req_str_or_addr)
+        else:
+            assert isinstance(req_str_or_addr, str)
+            # A local or VCS requirement will have an ampersand in it, and any other
+            # string with a path.sep in it will be an address spec.
+            if os.path.sep in req_str_or_addr and "@" not in req_str_or_addr:
+                specs.append(req_str_or_addr)
+            else:
+                req_strings.append(req_str_or_addr)
+    if specs:
+        addrs_from_specs = await Get(
+            Addresses,
+            UnparsedAddressInputs(
+                specs,
+                owning_address=None,
+                # TODO: Plumb the origin through in PexRequirements?
+                description_of_origin="User-specified requirements",
```

Comment:
That would probably be good, yea... maybe as an optional field (initially), at the very least? I would be less concerned about this, but particularly since heuristics are being used to distinguish addresses from requirements...

Reply:
BTW there may be a future cleanup (post removing tool lockfiles) where this only takes addresses, I'd have to dive more deeply to be sure.

Reply:
Switched to requiring the `//` prefix, for extra security (really I just check for a single `/`). 

Also added descriptions of origin.

Also add documentation, and apply this in the Pants repo. 

PTAL